### DISCUSSION TITLE
Refresh TextInputColumn value

### DISCUSSION THREAD

**Discussion by underdpt** [filamentphp/filament#5392] [Help]
Hello,

So I have a standalone table that's used to generate data to be saved to the database. In this case, I have some customer product orders and I want to generate new product orders for a supplier. I'm using a TextInputColumn to allow the user to specify how many units to order, and then save the new order.

The TextInputColumn state is saved to an array, so when the user finishes he types the order number and the order is created with all the products and quantities. So, I'm using `getStateUsing` and `updateStateUsing` methods to get the units to be ordered from that  array:

```php
            Columns\TextInputColumn::make('unidades_a_pedir')
                ->rules(fn (Articulo $record) => ['required', 'numeric', 'min:0', 'max:' . $record->unidades_a_servir])
                ->extraAttributes(fn (Articulo $record) => ['min' => 0, 'max' => $record->unidades_a_servir])
                ->type('number')
                ->step(1)
                ->getStateUsing(function (Articulo $record) {
                    return $this->values[$record->asin]['unidades'] ?? null;
                })
                ->updateStateUsing(function (int $state, Articulo $record) {
                    $this->values[$record->asin]['unidades'] = $state;

                    return $state;
                }),
```

I want to add some actions like 'Order all' and 'Order none', that change the array and  then trigger a refresh, like this:

```php
    public function pedirTodo(): void
    {
        $this->values = $this->getTableQuery()
            ->get()
            ->mapWithKeys(function (Articulo $articulo) {
                return [$articulo->asin => ['unidades' => (int) $articulo->unidades_a_servir]];
            })
            ->toArray();

        $this->emit('refresh');
    }
```

I can see that `getStateUsing()` is being called and it returns the correct amount, but the input is not refreshed (either continues blank or with the latest value manually set). In file `text-input-column.blade.php` the `$getState()` call also returns the correct amount.

Is this even possible, maybe a bug in the component?

**Comment by Congminh7799** [comment]
you solved it?

**Comment by Congminh7799** [comment]
u can overwrite text-input-columns.balde.php
add $state =  $getState to @php
and change x-model="state" to value="{{ $state }}"
it active good
![image](https://user-images.githubusercontent.com/106800650/221081992-9412ae3a-ef06-4d62-95b5-ae9ec5d8cc06.png)

**Reply by underdpt** [reply]
I did finally implement a simpler input column to achieve the desired result, but will try your solution because maybe it can be implemented directly in Filament.

Thanks!

**Reply by Congminh7799** [reply]
I had the same problem as you and it worked fine. 

Good luck!

**Reply by danharrin** [reply]
We're working on a fix for this internally

**Comment by danharrin** [comment]
All columns now refresh their value in v2.17.6.
